src/test/java/com/ibm/cloud/sdk/core/test/model/DiscriminatorSerializationTest.java

```diff
@@ -39,9 +39,9 @@
 import com.ibm.cloud.sdk.core.util.GsonSingleton;
```

Comment:
I changed this test class to use Builder classes when constructing the Animal-related model instances.
When Builder classes were introduced for dynamic models, it was done such that we are still generating the 
setter methods that we were previously generating for dynamic models as a way to maintain compatibility.
And that worked as we didn't "need" to change our dynamic model-related tests here in the core.
But that allowed this NPE bug to slip by :(